server/src/com/cloud/resourcelimit/ResourceLimitManagerImpl.java

```diff
@@ -886,60 +886,54 @@ public Long doInTransaction(TransactionStatus status) {
 
     @DB
     protected long recalculateAccountResourceCount(final long accountId, final ResourceType type) {
-        Long newCount = Transaction.execute(new TransactionCallback<Long>() {
-                @Override
-                public Long doInTransaction(TransactionStatus status) {
-                    Long newCount = null;
-                    lockAccountAndOwnerDomainRows(accountId, type);
```

Comment:
Shouldn't you maintain the lock account? I mean, if this method is called when a VM is being deployed, this can generate race conditions that will lead the table to be populated with inconsistent data.

Reply:
Aren't these aggregated values (of resources usage/allocated) used to control the deployment of new resources?

I thought they were, that is why I thought it is important to have them consistent at all time.